idea/data-storage/src/code/code.service.ts

```diff
@@ -38,41 +37,33 @@ export class CodeService {
       throw new CodeNotFound();
     }
 
-    if(code.meta === null) throw new MetadataNotFound();
+    if (code.meta === null) throw new MetadataNotFound();
 
     return code.meta;
   }
 
-  public async updateCodes(updateCodesInput: UpdateCodeInput[] | CodeChangedInput[]): Promise<Code[]> {
-    const updateCodes = [];
+  public async createCodes(codes: Code[]) {
+    try {
+      return this.codeRepository.save(codes);
+    } catch (error) {
+      this.logger.error('Update codes error');
+      console.log(error);
+    }
+  }
+
+  public async updateCodes(updateCodesInput: CodeChangedInput[]): Promise<Code[]> {
+    const codes = [];
```

Comment:
```suggestion
    const codes = updateCodesInput.map(async (updateCodeInput) => {
      const { id, genesis } = updateCodeInput;
      const code = await this.codeRepository.getByIdAndGenesis(id, genesis);

      code.expiration = updateCodeInput.expiration;
      code.status = updateCodeInput.status;
      return code;
    });
```

Reply:
What the point of this way? I think it's better to avoid async callbacks. In this case codes will be array of promises